src/couch/src/couch_db_updater.erl

```diff
@@ -1086,8 +1088,15 @@ copy_docs(Db, #db{fd = DestFd} = NewDb, MixedInfos, Retry) ->
         {NewRevTree, FinalAcc} = couch_key_tree:mapfold(fun
             (_Rev, #leaf{ptr=Sp}=Leaf, leaf, SizesAcc) ->
                 {Body, AttInfos} = copy_doc_attachments(Db, Sp, DestFd),
+                IsComp = couch_compress:is_compressed(Body, Compress),
+                EJsonBody = case IsComp of
```

Comment:
Ah, though I see that's based off the make_doc_summary code. The difference here is that we don't care what its compressed as, where as make_doc_summary is converting a possibly compressed document to a different compression scheme.